### PR TITLE
fix(crons): Proper sorting for Monitors list

### PR DESCRIPTION
**Pull request by rjo100**
Properly sorts monitors by putting `disabled` monitors at the end of the list. This allows us to figure out later where the `disabled` status should live on which object (monitor on monitorenv). Also properly sorts on the latest monitorenv check-in

SQL here:
```
SELECT "sentry_monitor"."id",
       "sentry_monitor"."guid",
       "sentry_monitor"."slug",
       "sentry_monitor"."organization_id",
       "sentry_monitor"."project_id",
       "sentry_monitor"."name",
       "sentry_monitor"."status",
       "sentry_monitor"."type",
       "sentry_monitor"."config",
       "sentry_monitor"."next_checkin",
       "sentry_monitor"."last_checkin",
       "sentry_monitor"."date_added",
       CASE
         WHEN "sentry_monitor"."status" = 1 THEN 4
         ELSE (SELECT CASE
                        WHEN U0."status" = 5 THEN 0
                        WHEN U0."status" = 6 THEN 1
                        WHEN U0."status" = 4 THEN 2
                        WHEN U0."status" = 0 THEN 3
                        WHEN U0."status" = 1 THEN 4
                        ELSE NULL
                      end AS "status_ordering"
               FROM   "sentry_monitorenvironment" U0
               WHERE  ( U0."environment_id" IN ( 1, 2 )
                        AND U0."monitor_id" = ( "sentry_monitor"."id" ) )
               ORDER  BY "status_ordering" ASC
               LIMIT  1)
       end        AS "sorting",
       (SELECT U0."last_checkin"
        FROM   "sentry_monitorenvironment" U0
        WHERE  ( U0."environment_id" IN ( 1, 2 )
                 AND U0."monitor_id" = ( "sentry_monitor"."id" ) )
        ORDER  BY U0."last_checkin" DESC
        LIMIT  1) AS "last_checkin_monitorenvironment"
FROM   "sentry_monitor"
WHERE  ( "sentry_monitor"."organization_id" = 4551830674931714
         AND "sentry_monitor"."project_id" IN ( 4551830674931716 )
         AND NOT ( "sentry_monitor"."status" IN ( 2, 3 ) ) ) 
```

### CODE REVIEW

File: src/sentry/monitors/endpoints/organization_monitors.py [hunk 1]
```diff
@@ -77,17 +77,31 @@ def get(self, request: Request, organization: Organization) -> Response:
         else:
             environments = list(Environment.objects.filter(organization_id=organization.id))
 
-        # sort monitors by top monitor environment
-        monitorenvironment_top_status = Subquery(
-            MonitorEnvironment.objects.filter(
-                monitor__id=OuterRef("id"), environment__in=environments
+        # sort monitors by top monitor environment, then by last check-in
```
Comment (evanpurkhiser) [inline src/sentry/monitors/endpoints/organization_monitors.py:80]: The comment here seems slightly outdated now no?

File: src/sentry/monitors/endpoints/organization_monitors.py [hunk 2]
```diff
@@ -77,17 +77,31 @@ def get(self, request: Request, organization: Organization) -> Response:
...
+        monitor_environments_query = MonitorEnvironment.objects.filter(
+            monitor__id=OuterRef("id"), environment__in=environments
+        )
+
+        queryset = queryset.annotate(
+            environment_status_ordering=Case(
+                When(status=MonitorStatus.DISABLED, then=Value(4)),
```
Comment (evanpurkhiser) [inline src/sentry/monitors/endpoints/organization_monitors.py:87]: Hardcoding the value 4 here is brittle, instead what we probably want to do is something like
```suggestion
                When(status=MonitorStatus.DISABLED, then=Value(len(DEFAULT_ORDERING) + 1)),
```

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitors.py:87]: We'll also want to change this to `ObjectStatus` probably

Comment (rjo100) [reply src/sentry/monitors/endpoints/organization_monitors.py:87]: yeah makes sense...I'd rather do all the status updates in one go

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitors.py]: Yeah sounds good

File: src/sentry/monitors/endpoints/organization_monitors.py (hunk 1 above)
Comment (rjo100) [reply src/sentry/monitors/endpoints/organization_monitors.py:80]: outdated how? it's slightly confusing but i can diagram it out

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitors.py:80]: It doesn't indicate that this is sorting disabled monitors to the bottom right

File: src/sentry/monitors/endpoints/organization_monitors.py [hunk 3]
```diff
@@ -77,17 +77,31 @@ def get(self, request: Request, organization: Organization) -> Response:
...
-            MonitorEnvironment.objects.filter(
-                monitor__id=OuterRef("id"), environment__in=environments
+        # sort monitors by top monitor environment, then by last check-in
+        monitor_environments_query = MonitorEnvironment.objects.filter(
+            monitor__id=OuterRef("id"), environment__in=environments
+        )
+
+        queryset = queryset.annotate(
+            environment_status_ordering=Case(
+                When(status=MonitorStatus.DISABLED, then=Value(len(DEFAULT_ORDERING) + 1)),
```
Comment (evanpurkhiser) [inline src/sentry/monitors/endpoints/organization_monitors.py:87]: The `+ 1` may actually be extraneous here

**Review by evanpurkhiser — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Let's follow up with the comment update. Want to unblock you on this

File: src/sentry/monitors/endpoints/organization_monitors.py (hunk 3 above)
Comment (rjo100) [reply src/sentry/monitors/endpoints/organization_monitors.py:87]: ooh right. will fix

**Pull request merged**